homeassistant/components/withings/common.py

```diff
@@ -280,58 +716,403 @@ async def update_sleep_summary(self) -> SleepGetSummaryResponse:
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._data_manager.is_available
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return self._unique_id
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit of measurement of this entity, if any."""
+        return self._attribute.unit_of_measurement
+
+    @property
+    def icon(self) -> str:
+        """Icon to use in the frontend, if any."""
+        return self._attribute.icon
+
+    @property
+    def state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes.
+
+        Implemented by component base class. Convention for attribute names
+        is lowercase snake_case.
+        """
+        return {"friendly_name": self._attribute.friendly_name}
+
+    @callback
+    def receive_data(self, state_data: StateData) -> None:
+        """Receive data from data dispatcher."""
+        if state_data.unique_id != self._unique_id:
+            return
+
+        self._state_data = state_data
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Register update dispatcher."""
+        self.async_on_remove(
+            async_dispatcher_connect(self.hass, DATA_UPDATED_SIGNAL, self.receive_data)
+        )
+
+
+async def async_init_data_manager(
+    hass: HomeAssistant, config_entry: ConfigEntry
 ) -> WithingsDataManager:
-    """Get a data manager for a config entry.
+    """Initialize the data manager for a config entry and oath2 implementation."""
+    profile = config_entry.data.get(const.PROFILE)
 
-    If the data manager doesn't exist yet, it will be
-    created and cached for later use.
-    """
-    entry_id = entry.entry_id
+    _LOGGER.debug("Creating withings data manager for profile: %s", profile)
+    data_manager = WithingsDataManager(
+        hass,
+        profile,
+        ConfigEntryWithingsApi(
+            hass=hass,
+            config_entry=config_entry,
+            implementation=await config_entry_oauth2_flow.async_get_config_entry_implementation(
+                hass, config_entry
+            ),
+        ),
+        config_entry.data["token"]["userid"],
+        WebhookConfig(
+            id=config_entry.data[CONF_WEBHOOK_ID],
+            url=config_entry.data[const.CONF_WEBHOOK_URL],
+            enabled=config_entry.data[const.USE_WEBHOOK],
+        ),
+    )
+
+    entry_id = config_entry.entry_id
 
     hass.data[const.DOMAIN] = hass.data.get(const.DOMAIN, {})
 
     domain_dict = hass.data[const.DOMAIN]
-    domain_dict[const.DATA_MANAGER] = domain_dict.get(const.DATA_MANAGER, {})
+    domain_dict[const.DATA_MANAGER] = domain_dict.get(const.DATA_MANAGER, OrderedDict())
```

Comment:
Normal dictionaries have been ordered since Python 3.6. `OrderedDict` is not necessary.